src/core/credGrainView.js

```diff
@@ -54,77 +60,133 @@ export type ParticipantsOptions = {|
+    }
+    return grainEarnedPerInterval;
   }
 
   withTimeScope(
-    startTimeMs: number,
-    endTimeMs: number
+    startTimeMs: TimestampMs,
+    endTimeMs: TimestampMs
   ): TimeScopedCredGrainView {
     return new TimeScopedCredGrainView(
-      this._credGraph,
-      this._ledger,
+      this._participants,
+      this._intervals,
       startTimeMs,
       endTimeMs
     );
   }
 
   intervals(): IntervalSequence {
-    return this._defaultTimeScope.intervals();
+    return cloneDeep(this._intervals);
   }
 
-  participant(id: IdentityId): ParticipantCredGrain | null {
-    return this._defaultTimeScope.participant(id);
+  participants(): $ReadOnlyArray<ParticipantCredGrain> {
+    return this._participants;
   }
 
-  participants(
-    options: ParticipantsOptions
-  ): $ReadOnlyArray<ParticipantCredGrain> {
-    return this._defaultTimeScope.participants(options);
+  participant(id: IdentityId): ParticipantCredGrain | null {
+    const _ = id;
+    throw new Error("not yet implemented");
   }
 }
 
 export class TimeScopedCredGrainView {
-  _credGraph: CredGraph;
-  _ledger: Ledger;
-  _startTimeMs: TimestampMs;
-  _endTimeMs: TimestampMs;
+  _participants: Array<ParticipantCredGrain>;
+  _intervals: IntervalSequence;
 
   constructor(
-    credGraph: CredGraph,
-    ledger: Ledger,
+    originalParticipants: $ReadOnlyArray<ParticipantCredGrain>,
+    originalIntervals: IntervalSequence,
     startTimeMs: TimestampMs,
     endTimeMs: TimestampMs
   ) {
-    this._credGraph = credGraph;
-    this._ledger = ledger;
-    this._startTimeMs = startTimeMs;
-    this._endTimeMs = endTimeMs;
+    let inclusiveStartIndex = originalIntervals.findIndex(
+      (interval) => startTimeMs <= interval.startTimeMs
+    );
+    if (inclusiveStartIndex === -1)
+      inclusiveStartIndex = originalIntervals.length;
+
+    const exclusiveEndIndex =
+      findLastIndex(
+        originalIntervals,
+        (interval) => interval.endTimeMs <= endTimeMs
+      ) + 1;
+
+    this._intervals = intervalSequence(
+      originalIntervals.slice(inclusiveStartIndex, exclusiveEndIndex)
+    );
+    this._participants = originalParticipants.map((participant) => {
+      const credPerInterval = participant.credPerInterval.slice(
+        inclusiveStartIndex,
+        exclusiveEndIndex
+      );
+      const grainEarnedPerInterval = participant.grainEarnedPerInterval.slice(
+        inclusiveStartIndex,
+        exclusiveEndIndex
+      );
+      return {
+        identity: participant.identity,
+        cred: credPerInterval.reduce((a, b) => a + b),
+        credPerInterval,
+        grainEarned: grainEarnedPerInterval.reduce((a, b) => G.add(a, b)),
+        grainEarnedPerInterval,
+      };
+    });
   }
 
-  participants(
-    options: ParticipantsOptions
-  ): $ReadOnlyArray<ParticipantCredGrain> {
-    const _ = options;
-    throw new Error("not yet implemented");
+  intervals(): IntervalSequence {
+    return cloneDeep(this._intervals);
```

Comment:
Same question as before for why we clone the intervals but not the participants.
Note if we were to be consistent (clone the participants too) that would make these APIs un-performant, which suggests to me we should take a different approach, e.g. using freezes.

Reply:
sure, will use freeze